File: python/03_fibonacci.py
# Python Program 3: Fibonacci Sequence
# Author: Lydia S. Makiwa

def fibonacci(n):
    """Returns a list of the first n Fibonacci numbers."""
    fib = [0, 1]
    for i in range(2, n):
        fib.append(fib[-1] + fib[-2])
    return fib[:n]

n = int(input("How many Fibonacci numbers? "))
print(f"First {n} Fibonacci numbers: {fibonacci(n)}")

File: c-programs/03_fibonacci.c
/* C Program 3: Fibonacci Series
   Author: Lydia S. Makiwa */

#include <stdio.h>

int main() {
    int n, a = 0, b = 1, next;
    printf("How many terms? ");
    scanf("%d", &n);
    printf("Fibonacci Series: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", a);
        next = a + b;
        a = b;
        b = next;
    }
    printf("\n");
    return 0;
}
